exo-go/src/app_setup/index_test.go

```diff
@@ -37,115 +40,154 @@ var _ = Describe("Setup", func() {
+			environment := dockerCompose.Services["exocom0.22.1"].Environment
+			Expect(environment["PORT"]).To(Equal("$EXOCOM_PORT"))
+			expectedServiceRoutes := []string{
+				`{"receives":["todo.create"],"role":"todo-service","sends":["todo.created"]}`,
+				`{"namespace":"mongo","receives":["mongo.list","mongo.create"],"role":"users-service","sends":["mongo.listed","mongo.created"]}`,
+				`{"receives":["todo.created"],"role":"html-server","sends":["todo.create"]}`,
+				`{"receives":["users.listed","users.created"],"role":"external-service","sends":["users.list","users.create"]}`,
+			}
+			for _, serviceRoute := range expectedServiceRoutes {
+				Expect(strings.Contains(environment["SERVICE_ROUTES"], serviceRoute))
+			}
+		})
+
+		It("should include exocom environment variables in internal services' environment", func() {
+			for _, serviceName := range internalServices {
+				environment := dockerCompose.Services[serviceName].Environment
+				Expect(environment["EXOCOM_HOST"]).To(Equal("exocom0.22.1"))
+				Expect(environment["EXOCOM_PORT"]).To(Equal("$EXOCOM_PORT"))
+			}
+		})
+
+		It("should generate a volume path for an external dependency that mounts a volume", func() {
+			Expect(len(dockerCompose.Services["mongo3.4.0"].Volumes)).NotTo(Equal(0))
+		})
+
+		It("should have the specified image and container names for the external service", func() {
+			serviceName := "external-service"
+			imageName := "originate/test-web-server"
+			Expect(dockerCompose.Services[serviceName].Image).To(Equal(imageName))
+			Expect(dockerCompose.Services[serviceName].ContainerName).To(Equal(serviceName))
+		})
+
+		It("should have the specified ports, volumes and environment variables for the external service", func() {
+			serviceName := "external-service"
+			ports := []string{"5000:5000"}
+			Expect(dockerCompose.Services[serviceName].Ports).To(Equal(ports))
+			Expect(len(dockerCompose.Services[serviceName].Volumes)).NotTo(Equal(0))
+			Expect(dockerCompose.Services[serviceName].Environment["EXTERNAL_SERVICE_HOST"]).To(Equal("external-service0.1.2"))
+			Expect(dockerCompose.Services[serviceName].Environment["EXTERNAL_SERVICE_PORT"]).To(Equal("$EXTERNAL_SERVICE_PORT"))
+		})
+
+		It("should have the ports for the external dependency defined in application.yml", func() {
+			serviceName := "mongo3.4.0"
+			ports := []string{"4000:4000"}
+			Expect(dockerCompose.Services[serviceName].Ports).To(Equal(ports))
+			Expect(len(dockerCompose.Services[serviceName].Volumes)).NotTo(Equal(0))
+		})
 	})
 
-	It("should include external dependencies as dependencies", func() {
-		exists := util.DoesStringArrayContain(dockerCompose.Services["todo-service"].DependsOn, "mongo3.4.0")
-		Expect(exists).To(Equal(true))
+	var _ = Describe("Dockerfile", func() {
+		It("should update the dockerfiles of internal services with the commands defined in service.yml and not modify the commands in the original dockerfiles", func() {
```

Comment:
I can see where you are testing "should update the dockerfiles of internal services with the commands defined in service.yml" but I don't see anything testing "not modify the commands in the original dockerfile"